flink-ml-lib/src/main/java/org/apache/flink/ml/classification/knn/KnnModel.java

```diff
@@ -0,0 +1,378 @@
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(input),
+                        Collections.singletonMap(broadcastKey, model),
+                        inputList -> {
+                            DataStream inoutData = inputList.get(0);
+                            return inoutData.transform(
+                                    "mapFunc",
+                                    TableUtils.getRowTypeInfo(outputSchema),
+                                    new PredictOperator(broadcastKey, getK(), getFeaturesCol()));
+                        });
+
+        return new Table[] {tEnv.fromDataStream(output)};
+    }
+
+    /** @return Parameters for algorithm. */
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return this.params;
+    }
+
+    /**
+     * Saves model data.
+     *
+     * @param path Path to save.
+     */
+    @Override
+    public void save(String path) throws IOException {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) modelData[0]).getTableEnvironment();
+
+        String dataPath = ReadWriteUtils.getDataPath(path);
+        FileSink<Row> sink =
+                FileSink.forRowFormat(new Path(dataPath), new KnnModelData.ModelDataEncoder())
+                        .withRollingPolicy(OnCheckpointRollingPolicy.build())
+                        .withBucketAssigner(new BasePathBucketAssigner<>())
+                        .build();
+        tEnv.toDataStream(modelData[0]).sinkTo(sink);
+        ReadWriteUtils.saveMetadata(this, path);
+    }
+
+    /**
+     * Loads model data from path.
+     *
+     * @param env Stream execution environment.
+     * @param path Model path.
+     * @return Knn model.
+     */
+    public static KnnModel load(StreamExecutionEnvironment env, String path) throws IOException {
+        StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+        KnnModel retModel = ReadWriteUtils.loadStageParam(path);
+
+        Source<Row, ?, ?> source =
+                FileSource.forRecordStreamFormat(
+                                new KnnModelData.ModelDataStreamFormat(),
+                                ReadWriteUtils.getDataPaths(path))
+                        .build();
+        DataStream<Row> modelDataStream =
+                env.fromSource(source, WatermarkStrategy.noWatermarks(), "data");
+        retModel.modelData =
+                new Table[] {tEnv.fromDataStream(modelDataStream, KnnModelData.getModelSchema())};
+        return retModel;
+    }
+
+    /** This operator loads model data and predicts result. */
+    private static class PredictOperator
+            extends AbstractUdfStreamOperator<Row, AbstractRichFunction>
+            implements OneInputStreamOperator<Row, Row> {
+
+        private boolean firstEle = true;
+        private final String featureCol;
+        private transient KnnModelData modelData;
+        private final Integer topN;
```

Comment:
nits: we use `int` instead of `Integer` when possible. Could we use `int` here?

Reply:
OK